pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -838,26 +838,27 @@ func (r *ReconcileClusterDeployment) reconcileInstallingClusterInstall(cd *hivev
 	return r.reconcileExistingInstallingClusterInstall(cd, logger)
 }
 
-func dnsZoneNotReadyMaybeTimedOut(cd *hivev1.ClusterDeployment, logger log.FieldLogger) (string, string, time.Duration) {
+func dnsZoneNotReadyMaybeTimedOut(cd *hivev1.ClusterDeployment, logger log.FieldLogger) (string, string, time.Duration, bool) {
```

Comment:
Time to add a docstring to this func describing the inputs and outputs, noting that it has side effects on the `cd`. On that note, it sure would be nice if the bool return cleanly indicated that the cd was changed *at all* (indicating that the caller should do a Status().Update()). I get that the caller only wants to bump the metric if the `ProvisionStopped` condition has changed... but maybe we handle that a different way -- see below.